src/common/datasource/src/file_format/parquet.rs

```diff
@@ -174,123 +174,47 @@ impl ArrowWriterCloser for ArrowWriter<SharedBuffer> {
-/// storage by chunks to reduce memory consumption.
-pub struct BufferedWriter {
-    inner: InnerBufferedWriter,
-}
-
-type InnerBufferedWriter = LazyBufferedWriter<
-    Compat<object_store::FuturesAsyncWriter>,
-    ArrowWriter<SharedBuffer>,
-    impl Fn(String) -> BoxFuture<'static, Result<Compat<object_store::FuturesAsyncWriter>>>,
->;
-
-impl BufferedWriter {
-    fn make_write_factory(
-        store: ObjectStore,
-        concurrency: usize,
-    ) -> impl Fn(String) -> BoxFuture<'static, Result<Compat<object_store::FuturesAsyncWriter>>>
-    {
-        move |path| {
-            let store = store.clone();
-            Box::pin(async move {
-                store
-                    .writer_with(&path)
-                    .concurrent(concurrency)
-                    .chunk(DEFAULT_WRITE_BUFFER_SIZE.as_bytes() as usize)
-                    .await
-                    .map(|v| v.into_futures_async_write().compat_write())
-                    .context(error::WriteObjectSnafu { path })
-            })
-        }
-    }
-
-    pub async fn try_new(
-        path: String,
-        store: ObjectStore,
-        arrow_schema: SchemaRef,
-        props: Option<WriterProperties>,
-        buffer_threshold: usize,
-        concurrency: usize,
-    ) -> error::Result<Self> {
-        let buffer = SharedBuffer::with_capacity(buffer_threshold);
-
-        let arrow_writer = ArrowWriter::try_new(buffer.clone(), arrow_schema.clone(), props)
-            .context(error::WriteParquetSnafu { path: &path })?;
-
-        Ok(Self {
-            inner: LazyBufferedWriter::new(
-                buffer_threshold,
-                buffer,
-                arrow_writer,
-                &path,
-                Self::make_write_factory(store, concurrency),
-            ),
-        })
-    }
-
-    /// Write a record batch to stream writer.
-    pub async fn write(&mut self, arrow_batch: &RecordBatch) -> error::Result<()> {
-        self.inner.write(arrow_batch).await
-    }
-
-    /// Close parquet writer.
-    ///
-    /// Return file metadata and bytes written.
-    pub async fn close(self) -> error::Result<(FileMetaData, u64)> {
-        self.inner.close_with_arrow_writer().await
-    }
-}
-
 /// Output the stream to a parquet file.
 ///
 /// Returns number of rows written.
 pub async fn stream_to_parquet(
     mut stream: SendableRecordBatchStream,
     store: ObjectStore,
     path: &str,
-    threshold: usize,
     concurrency: usize,
 ) -> Result<usize> {
     let write_props = WriterProperties::builder()
         .set_compression(Compression::ZSTD(ZstdLevel::default()))
         .build();
     let schema = stream.schema();
-    let mut buffered_writer = BufferedWriter::try_new(
-        path.to_string(),
-        store,
-        schema,
-        Some(write_props),
-        threshold,
-        concurrency,
-    )
-    .await?;
+    let inner_writer = store
+        .writer_with(path)
```

Comment:
We must invoke the `chunk` method to avoid the `EntityTooSmall` error from S3. BTW, we can utilize the tests introduced in #4105, which ensure there are no `EntityTooSmall ` errors returned from the MinIO during the flush and compaction operations.

Reply:
> We must invoke the `chunk` method to avoid the `EntityTooSmall` errors from S3. BTW, we can utilize the tests introduced in #4105, which ensure there are no `EntityTooSmall ` errors returned from the MinIO during the flush and compaction operations.

fixed in 1a2bd1020